com.kms.cura.dal/src/com/kms/cura/dal/database/AppointmentDatabaseHelper.java

```diff
@@ -0,0 +1,117 @@
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.FacilityEntity;
+import com.kms.cura.entity.user.DoctorUserEntity;
+import com.kms.cura.entity.user.PatientUserEntity;
+
+public class AppointmentDatabaseHelper extends DatabaseHelper {
+
+	public AppointmentDatabaseHelper() throws ClassNotFoundException, SQLException {
+		super();
+	}
+
+	@Override
+	protected Entity getEntityFromResultSet(ResultSet resultSet) throws SQLException, ClassNotFoundException {
+		return null;
+	}
+
+	protected AppointmentEntity geAppointmentEntityFromResultSet(ResultSet resultSet,PatientUserEntity patientUserEntity, DoctorUserEntity doctorUserEntity)
+			throws SQLException, ClassNotFoundException {
+		DoctorUserEntity doctor = null;
+		PatientUserEntity patient = null;
+		if(doctorUserEntity == null){
+			doctor = (DoctorUserEntity) new DoctorUserDatabaseHelper().querybyID(resultSet.getInt(AppointmentColumn.DOCTOR_ID.getColumnName()));
+		}
+		if(patientUserEntity == null){
+			patient = (PatientUserEntity) new PatientUserDatabaseHelper().querybyID(resultSet.getInt(AppointmentColumn.PATIENT_ID.getColumnName()));
+		}
+		return new AppointmentEntity(patient, doctor,
+				(FacilityEntity) FacilityDAL.getInstance().getByID(
+						resultSet.getInt(AppointmentColumn.FACILITY_ID.getColumnName()), new FacilityDatabaseHelper()),
+				resultSet.getDate(AppointmentColumn.APPT_DAY.getColumnName()),
+				resultSet.getTime(AppointmentColumn.START_TIME.getColumnName()),
+				resultSet.getTime(AppointmentColumn.END_TIME.getColumnName()),
+				resultSet.getInt(AppointmentColumn.STATUS.getColumnName()));
+	}
+
+	public List<AppointmentEntity> getAppointment(AppointSearchEntity criteria, PatientUserEntity patientUserEntity,
+			DoctorUserEntity doctorUserEntity) throws SQLException, ClassNotFoundException {
+		List<AppointmentEntity> listAppts = new ArrayList<>();
+		AppointmentEntity entity = criteria.getAppointmentEntity();
+		ResultSet rs = null;
+		PreparedStatement stmt = null;
+		try {
+			stmt = getAppointmentQuery(entity);
+			rs = stmt.executeQuery();
+			if (rs != null) {
+				while (rs.next()) {
+					listAppts.add(geAppointmentEntityFromResultSet(rs, patientUserEntity, doctorUserEntity));
+				}
+			}
+			return listAppts;
+		} finally {
+			if (rs != null) {
+				rs.close();
+			}
+			if (stmt != null) {
+				stmt.close();
+			}
+			if (con != null) {
+				closeConnection();
+			}
+		}
+	}
+
+	private PreparedStatement getAppointmentQuery(AppointmentEntity entity) throws SQLException {
+		StringBuilder builder = new StringBuilder();
+		PreparedStatement stmt = null;
+		builder.append("SELECT * FROM ");
+		builder.append(AppointmentColumn.TABLE_NAME);
+		builder.append(" WHERE ");
+		if (entity.getPatientUserEntity() != null) {
+			builder.append(AppointmentColumn.PATIENT_ID.getColumnName());
+			builder.append(" = ");
+			builder.append(entity.getPatientUserEntity().getId());
+		}
+		if (entity.getDoctorUserEntity() != null) {
+			builder.append(" AND ");
```

Comment:
There maybe a leak here. If entity.getPatientUserEntity() is null and entity.getDoctorUserEntity() is not null, this query will be like "...WHERE AND ..." and would be broken.